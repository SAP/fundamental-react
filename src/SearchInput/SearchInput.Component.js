import { SearchInput } from './SearchInput';
import { Description, DocsText, DocsTile, Header, Import, Properties, Separator } from '../_playground';
import React, { Component } from 'react';

export class SearchInputComponent extends Component {
    searchData = [
        { text: 'apple', callback: () => alert('apple') },
        { text: 'apricot', callback: () => alert('apricot') },
        { text: 'banana', callback: () => alert('banana') },
        { text: 'blueberry', callback: () => alert('blueberry') },
        { text: 'blackberry', callback: () => alert('blackberry') },
        { text: 'calabash', callback: () => alert('calabash') },
        { text: 'clementines', callback: () => alert('clementines') },
        { text: 'kiwi', callback: () => alert('kiwi') },
        { text: 'orange', callback: () => alert('orange') }
    ];

    constructor(props) {
        super(props);
        this.state = {
            inputValue: ''
        };
    }

    getInputValue(value) {
        this.setState({
            inputValue: value
        });

        setTimeout(
            function() {
                alert('You just set the inputValue of the state to ' + this.state.inputValue);
            }.bind(this),
            100
        );
    }

    onChangeCallback() {
        alert('Your custom onChange function is fired!');
    }

    searchInputCode = `<SearchInput
    placeholder='Enter a fruit'
    searchList={this.searchData}
    onEnter={term => this.getInputValue(term)} />

    
<SearchInput
    placeholder='Enter a fruit'
    noSearchBtn
    onChange={this.onChangeCallback} />

    
<SearchInput
    compact
    placeholder='Enter a fruit'
    searchList={this.searchData}
    onEnter={term => this.getInputValue(term)} />


************************************ Data ************************************

getInputValue(value) {
    this.setState({
        inputValue: value
    });

    setTimeout(
        function() {
            alert('You just set the inputValue of the state to ' + this.state.inputValue);
        }.bind(this),
        1000
    );
}

searchData = [
    { text: 'apple', callback: () => alert('apple') },
    { text: 'apricot', callback: () => alert('apricot') },
    { text: 'banana', callback: () => alert('banana') },
    { text: 'blueberry', callback: () => alert('blueberry') },
    { text: 'blackberry', callback: () => alert('blackberry') },
    { text: 'calabash', callback: () => alert('calabash') },
    { text: 'clementines', callback: () => alert('clementines') },
    { text: 'kiwi', callback: () => alert('kiwi') },
    { text: 'orange', callback: () => alert('orange') }
];`;

    render() {
        return (
            <div>
                <Header>Search Input</Header>
                <Description />
                <Import module='SearchInput' path='/fundamental-react/src/' />

                <Separator />

                <Properties
                    properties={[
                        {
                            name: 'placeholder',
                            description: 'string - The text to use as placeholder when no text is entered.'
                        },
                        {
                            name: 'searchList',
                            description: 'array - Collection of items to display in auto-complete list.'
                        },
                        {
                            name: 'onEnter',
                            description: 'func - Method to execute by pressing the Enter key.'
                        },
                        {
                            name: 'noSearhBtn',
                            description: 'bool - set to true to hide the Search button.'
                        },
                        {
                            name: 'onChange',
                            description:
                                'func - a custom implementation of onChange. If not specified, the default behavior will be applied.'
                        },
                        {
                            name: 'compact',
                            description: 'bool - Set to true to enable compact mode.'
                        }
                    ]}
                    type='Inputs' />

                <Separator />

                <DocsTile>
                    <div>
                        <SearchInput
                            onEnter={term => this.getInputValue(term)}
                            placeholder='Enter a fruit'
                            searchList={this.searchData} />
                        <br />
                        <SearchInput
                            noSearchBtn
                            onChange={this.onChangeCallback}
                            placeholder='Enter a fruit' />
                        <br />
                        <SearchInput
                            compact
                            onEnter={term => this.getInputValue(term)}
                            placeholder='Enter a fruit'
                            searchList={this.searchData} />
                    </div>
                </DocsTile>
                <DocsText>{this.searchInputCode}</DocsText>

                <Separator />
            </div>
        );
    }
}
