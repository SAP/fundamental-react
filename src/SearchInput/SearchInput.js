import Button from '../Button/Button';
import classnamesBind from 'classnames/bind';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import { FORM_MESSAGE_TYPES } from '../utils/constants';
import FormInput from '../Forms/FormInput';
import FormMessage from '../Forms/_FormMessage';
import InputGroup from '../InputGroup/InputGroup';
import keycode from 'keycode';
import Menu from '../Menu/Menu';
import Popover from '../Popover/Popover';
import PropTypes from 'prop-types';
import withStyles from '../utils/withStyles';
import React, { useEffect, useState } from 'react';
import styles from 'fundamental-styles/dist/input-group.css';

const classnames = classnamesBind.bind(styles);

const SearchInput = React.forwardRef( ({
    className,
    compact,
    cssNamespace,
    disabled,
    formMessageProps,
    inputProps,
    inputGroupAddonProps,
    inputGroupProps,
    inShellbar,
    listProps,
    localizedText,
    noSearchBtn,
    onChange,
    onEnter,
    onSelect,
    placeholder,
    popoverProps,
    readOnly,
    searchList,
    subStringSearch,
    searchBtnProps,
    validationState,
    ...rest
}, ref) => {
    const [isExpanded, setIsExpanded ] = useState(false);
    const [searchExpanded, setSearchExpanded] = useState(false);
    const [value, setValue] = useState(inputProps?.value || '');

    const filterList = (list, query) => {
        return subStringSearch ? list.filter((item) => {
            return item.text.toLowerCase().includes(query.toLowerCase());
        }) : list.filter((item) => item.text.toLowerCase().startsWith(query.toLowerCase()));
    };

    const handleKeyPress = event => {
        if (keycode(event) === 'enter') {
            onEnter(value);
        }
    };

    const handleListItemClick = (event, item) => {
        setValue(item.text);
        setIsExpanded(false);
        setSearchExpanded(false);
        item?.callback();
        onSelect(event, item);
    };

    const handleChange = event => {
        let filteredResult;
        if (searchList) {
            filteredResult = filterList(searchList, event.target.value);
        }
        setValue(event.target.value);
        setIsExpanded(true);
        onChange(event, filteredResult);
    };

    const handleClick = () => {
        if (!readOnly) {
            setIsExpanded(!isExpanded);
        }
    };

    const handleClickOutside = () => {
        setIsExpanded(false);
        setSearchExpanded(false);
    };

    const handleEsc = event => {
        if (
            (event.keyCode === 27 && isExpanded === true) ||
            (event.keyCode === 27 && searchExpanded === true)
        ) {
            setIsExpanded(false);
            setSearchExpanded(false);
            setValue('');
        }
    };

    useEffect(() => {
        document.addEventListener('keydown', handleEsc, false);

        return () => {
            document.removeEventListener('keydown', handleEsc, false);
        };
    });


    let inputGroupClasses = inputGroupProps && inputGroupProps.className;

    inputGroupClasses = !inShellbar ? classnames(
        inputGroupClasses,
        `${cssNamespace}-input-group--control`,
        {
            [`is-${validationState?.state}`]: validationState?.state
        }
    ) : inputGroupClasses;
    let filteredResult = value && searchList ? filterList(searchList, value) : searchList;
    const popoverBody = (
        <Menu>
            <Menu.List {...listProps}>
                {filteredResult && filteredResult.length > 0 ? (
                    filteredResult.map((item, index) => {
                        return (
                            subStringSearch ? (<Menu.Item
                                key={index}
                                onClick={(e) => handleListItemClick(e, item)}>
                                {item.text}
                            </Menu.Item>) :
                                (
                                    <Menu.Item
                                        key={index}
                                        onClick={(e) => handleListItemClick(e, item)}>
                                        <strong>{value}</strong>
                                        {value && value.length
                                            ? item.text.substring(value.length)
                                            : item.text}
                                    </Menu.Item>
                                )
                        );
                    })
                ) : (
                    <Menu.Item>No result</Menu.Item>
                )}
            </Menu.List>
        </Menu>
    );

    const inputGroup = (
        <InputGroup
            {...inputGroupProps}
            className={inputGroupClasses}
            compact={compact}
            disabled={disabled}
            readOnly={readOnly}
            // need to get the styling into the input group, but without creating a duplicate popover
            validationState={{ state: validationState?.state, text: '' }}>
            <FormInput
                {...inputProps}
                disabled={disabled}
                onChange={handleChange}
                onClick={handleClick}
                onKeyPress={handleKeyPress}
                placeholder={placeholder}
                readOnly={readOnly}
                value={value} />

            { !(noSearchBtn || readOnly) && (
                <InputGroup.Addon {...inputGroupAddonProps} isButton>
                    <Button {...searchBtnProps}
                        aria-label={localizedText.searchBtnLabel}
                        disabled={disabled}
                        glyph='search'
                        onClick={handleClick}
                        option='transparent' />
                </InputGroup.Addon>
            )}
        </InputGroup>
    );

    return (
        <div
            {...rest}
            className={className}
            ref={ref}>
            <Popover
                {...popoverProps}
                body={
                    (<>
                        {validationState &&
                            <FormMessage
                                {...formMessageProps}
                                type={validationState.state}>
                                {validationState.text}
                            </FormMessage>
                        }
                        {popoverBody}
                    </>)}
                control={inputGroup}
                disableKeyPressHandler
                disabled={readOnly}
                noArrow
                onClickOutside={handleClickOutside}
                widthSizingType='minTarget' />
        </div>
    );
});

SearchInput.displayName = 'SearchInput';

SearchInput.propTypes = {
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Set to **true** to enable compact mode */
    compact: PropTypes.bool,
    /** Set to **true** to mark component as disabled and make it non-interactive */
    disabled: PropTypes.bool,
    /** Additional props to be spread to the FormMessage component */
    formMessageProps: PropTypes.object,
    /** Props to be spread to the InputGroupAddon component */
    inputGroupAddonProps: PropTypes.object,
    /** Props to be spread to the InputGroup component */
    inputGroupProps: PropTypes.object,
    /** Additional props to be spread to the `<input>` element */
    inputProps: PropTypes.object,
    inShellbar: PropTypes.bool,
    /** Additional props to be spread to the `<ul>` element */
    listProps: PropTypes.object,
    /** Localized text to be updated based on location/language */
    localizedText: CustomPropTypes.i18n({
        searchBtnLabel: PropTypes.string
    }),
    /** Set to **true** to render without a search button */
    noSearchBtn: PropTypes.bool,
    /** Localized placeholder text of the input */
    placeholder: PropTypes.string,
    /** Additional props to be spread to the Popover component */
    popoverProps: PropTypes.object,
    /** Set to **true** to mark component as readonly */
    readOnly: PropTypes.bool,
    /** Additional props to be spread to the search `<button>` element */
    searchBtnProps: PropTypes.object,
    /** Collection of items to display in the dropdown list */
    searchList: PropTypes.arrayOf(
        PropTypes.shape({
            text: PropTypes.string.isRequired,
            callback: PropTypes.func
        })
    ),
    /** enable substring search */
    subStringSearch: PropTypes.bool,
    /** An object identifying a validation message.  The object will include properties for `state` and `text`; _e.g._, \`{ state: \'warning\', text: \'This is your last warning\' }\` */
    validationState: PropTypes.shape({
        /** State of validation: 'error', 'warning', 'information', 'success' */
        state: PropTypes.oneOf(FORM_MESSAGE_TYPES),
        /** Text of the validation message */
        text: PropTypes.string
    }),
    /**
     * Callback function; triggered when a change event is fired on the underlying `<input>`.
     *
     * @param {SyntheticEvent} event - React's original SyntheticEvent. See https://reactjs.org/docs/events.html.
     * @param {Object[]} matches - array of objects from searchList that match the search string.
     * @returns {void}
    */
    onChange: PropTypes.func,
    /**
     * Callback function; triggered when the user hits the `enter` key in the text `<input>`.
     *
     * @param {string} text - current string value of input field.
     * @returns {void}
     * */
    onEnter: PropTypes.func,
    /**
     * Callback function; triggered when user clicks on an option.
     *
     * @param {SyntheticEvent} event - React's original SyntheticEvent. See https://reactjs.org/docs/events.html.
     * @param {Object} data - selected object from searchList
     * @returns {void}
     * */
    onSelect: PropTypes.func
};

SearchInput.defaultProps = {
    localizedText: {
        searchBtnLabel: 'Search'
    },
    onChange: () => { },
    onEnter: () => { },
    onSelect: () => { }
};

export default withStyles(SearchInput);
