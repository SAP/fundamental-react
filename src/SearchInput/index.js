export { default as SearchInput } from './SearchInput.js';
