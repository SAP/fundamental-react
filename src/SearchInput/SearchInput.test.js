import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import React from 'react';
import SearchInput from './SearchInput';

describe('<SearchInput />', () => {
    const searchInput = 'input[type="text"].fd-input';

    const getInputValue = value => {
        return value;
    };

    const searchData = [
        { text: 'apple', callback: jest.fn() },
        { text: 'apricot', callback: jest.fn() },
        { text: 'banana', callback: jest.fn() },
        { text: 'blueberry', callback: jest.fn() },
        { text: 'blackberry', callback: jest.fn() },
        { text: 'calabash', callback: jest.fn() },
        { text: 'clementines', callback: jest.fn() },
        { text: 'kiwi', callback: jest.fn() },
        { text: 'orange', callback: jest.fn() }
    ];


    const defaultSearchInputOnEnter = jest.fn();
    const defaultSearchInputOnChange = jest.fn();
    const defaultSearchInput = (
        <SearchInput
            className='blue'
            onChange={defaultSearchInputOnChange}
            onEnter={defaultSearchInputOnEnter}
            placeholder='Enter a fruit'
            searchList={searchData} />
    );

    let wrapper = null;
    beforeEach(() => {
        wrapper = mount(defaultSearchInput);
    });

    afterEach(() => {
        if (wrapper) {
            wrapper.unmount();
            wrapper = null;
        }
    });

    test('check for enter key press on search input', () => {
        act(() => {
            // enter text into search input
            wrapper
                .find(searchInput)
                .simulate('change', { target: { value: searchData[0].text } });
        });

        act(() => {
            // press enter key
            wrapper.find(searchInput).simulate('keypress',
                {
                    keyCode: 13
                }
            );
        });

        expect(defaultSearchInputOnEnter).toHaveBeenLastCalledWith(searchData[0].text);
    });

    test('click outside search input to close list', () => {
        let event = new MouseEvent('mousedown', {});

        act(() => {
            // outside click, search list not shown
            document.dispatchEvent(event);
        });

        act(() => {
            // enter text into search input
            wrapper
                .find(searchInput)
                .simulate('change', { target: { value: searchData[0].text } });
        });

        act(() => {
            // click outside to close list
            document.dispatchEvent(event);
        });

        expect(document.querySelectorAll('ul').length).toBe(0);
    });

    test('show/hide auto complete list', () => {
        act(() => {
            // click in search box to show
            wrapper.find(searchInput).simulate('click');
        });
        expect(document.querySelectorAll('ul').length).toBe(1);

        act(() => {
            // click in search box to hide
            wrapper.find(searchInput).simulate('click');
        });

        expect(document.querySelectorAll('ul').length).toBe(0);
    });

    test('check for enter key press on search input without autocomplete', () => {

        const noListSearchInputOnEnter = jest.fn();
        const noListSearchInput = (
            <SearchInput
                onEnter={noListSearchInputOnEnter}
                placeholder='Enter a fruit' />
        );

        const internalWrapper = mount(noListSearchInput);

        act(() => {
            // click in search box
            internalWrapper.find(searchInput).simulate('click');
        });

        act(() => {
            // enter text into search input
            internalWrapper
                .find(searchInput)
                .simulate('change', { target: { value: searchData[2].text } });
        });

        act(() => {
            // press enter key
            internalWrapper.find(searchInput).simulate('keypress',
                {
                    keyCode: 13
                }
            );
        });

        expect(noListSearchInputOnEnter).toHaveBeenLastCalledWith(searchData[2].text);

        internalWrapper.unmount();
    });

    test('click on result in autocomplete', () => {
        act(() => {
            // click in search box to show
            wrapper.find(searchInput).simulate('click');
        });

        expect(document.querySelectorAll('ul').length).toBe(1);

        act(() => {
            // select option
            document
                .querySelectorAll('.fd-menu__link')[0]
                .click();
        });

        act(() => {
            // click in search box to hide
            wrapper.find(searchInput).simulate('click');
        });

        expect(document.querySelectorAll('ul').length).toBe(0);
    });

    test('check search executed on search button click', () => {
        act(() => {
            wrapper
                .find(searchInput)
                .simulate('change', { target: { value: searchData[0].text } });
        });

        // check if searchTerm state is updated
        expect(defaultSearchInputOnChange).toHaveBeenLastCalledWith(
            expect.objectContaining({ target: { value: searchData[0].text } }),
            expect.arrayContaining([expect.objectContaining({ text: searchData[0].text })])
        );

        act(() => {
            wrapper.find('.fd-button--transparent').simulate('click');
        });

        expect(defaultSearchInputOnChange).toHaveBeenLastCalledWith(
            expect.objectContaining({ target: { value: searchData[0].text } }),
            expect.arrayContaining([expect.objectContaining({ text: searchData[0].text })])
        );
    });

    test('pressing Esc key to close search list', () => {
        act(() => {
            // click in search box to show
            wrapper.find(searchInput).simulate('click');
        });

        expect(document.querySelectorAll('ul').length).toBe(1);

        act(() => {
            // handle esc key
            let event = new KeyboardEvent('keydown', { keyCode: 27 });
            document.dispatchEvent(event);
        });

        expect(document.querySelectorAll('ul').length).toBe(0);
    });


    describe('onChange handler', () => {
        test('calling parent onChange event', () => {
            const onChangeHandler = jest.fn();

            const searchOnChange = (
                <SearchInput
                    onChange={onChangeHandler}
                    onEnter={term => getInputValue(term)}
                    placeholder='Enter a fruit'
                    searchList={searchData} />
            );
            const internalWrapper = mount(searchOnChange);

            act(() => {
                // enter text into search input
                internalWrapper
                    .find(searchInput)
                    .simulate('change', { target: { value: searchData[0].text } });
            });

            expect(onChangeHandler).toHaveBeenLastCalledWith(
                expect.objectContaining({ target: { value: searchData[0].text } }),
                expect.arrayContaining([expect.objectContaining({ text: searchData[0].text })])
            );

            internalWrapper.unmount();
        });

        test('should dispatch the onChange callback with the event', () => {
            let f = jest.fn();
            const element = mount(<SearchInput onChange={f} />);

            act(() => {
                element.find('input').simulate('change');
            });

            expect(f).toHaveBeenCalledTimes(1);
            element.unmount();
        });


        test('subString Search', () => {
            let internalWrapper;

            act(() => {
                internalWrapper = mount(
                    <SearchInput
                        placeholder='Enter a value'
                        searchList={[
                            { text: 'who is a supplier user?' },
                            { text: 'who is a buyer user?' },
                            { text: 'who is a worker user?' },
                            { text: 'how to change the pin?' },
                            { text: 'how to set the pin?' }
                        ]}
                        subStringSearch />
                );
            });

            act(() => {
                internalWrapper.find(searchInput).simulate('click');
            });

            act(() => {
                internalWrapper.find(searchInput).simulate('change', { target: { value: 'supplier' } });
            });

            expect(document.querySelectorAll('li').length).toBe(1);

            internalWrapper.unmount();

        });
    });

    describe('Internal component prop spreading', () => {
        let setup = (props) => mount(<SearchInput searchList={searchData} {...props} />);
        afterEach(() => {
            document.body.innerHTML = '';
        });

        const getFormMessage = () => document.body.querySelector('.fd-popover__popper > div > .fd-form-message');

        test('should spread popoverProps to Popover', () => {
            const internalWrapper = setup({
                popoverProps: { 'data-sample': 'Sample', className: 'wonderful-styles' }
            });

            const popover = internalWrapper.find('.fd-popover').first().getDOMNode();

            expect(
                popover.attributes['data-sample'].value
            ).toBe('Sample');

            expect(
                popover.classList
            ).toContain('wonderful-styles');
        });

        test('should spread formMessageProps to FormMessage', async() => {
            let internalWrapper;
            await act(async() => {
                internalWrapper = setup({
                    formMessageProps: { 'data-sample': 'Sample', className: 'wonderful-styles' },
                    validationState: {
                        state: 'error',
                        text: 'Test validation state'
                    }
                });
            });

            await act(() => {
                internalWrapper.find(searchInput).simulate('click');
            });

            const messageNode = getFormMessage();

            expect(
                messageNode.attributes['data-sample'].value
            ).toBe('Sample');

            expect(
                messageNode.classList
            ).toContain('wonderful-styles');
        });
    });

    describe('Prop spreading', () => {

        beforeEach(() => {
            document.body.innerHTML = '';

        });

        test('should allow props to be spread to the SearchInput component', () => {
            const element = mount(<SearchInput data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the SearchInput component\'s input element', () => {
            let element = mount(<SearchInput inputProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('input').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');

            element = mount(<SearchInput inShellbar inputProps={{ 'data-sample': 'Sample1' }} />);

            expect(
                element.find('input').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample1');
        });

        test('should allow props to be spread to the SearchInput component\'s button element', () => {
            let element = mount(<SearchInput searchBtnProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('button').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');

            element = mount(<SearchInput inShellbar searchBtnProps={{ 'data-sample': 'Sample1' }} />);

            expect(
                element.find('button').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample1');
        });

        test('should allow props to be spread to the SearchInput component\'s ul element', () => {
            const element = mount(<SearchInput listProps={{ 'data-sample': 'Sample' }} searchList={searchData} />);

            element.find('.fd-input').simulate('click');

            expect(
                element.find('ul').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');

        });

        const searchDataNew = [
            { text: 'peaches' },
            { text: 'pear' }
        ];

        test('should allow props list to be changed after creation', () => {
            class Test extends React.Component {
                constructor(props) {
                    super(props);
                    this.state = {
                        list: searchData
                    };
                }

                handleChange = (e) => {
                    if (e.target.value === 'pe') {
                        this.setState({
                            list: searchDataNew
                        });
                    }
                }
                render = () => (<SearchInput onChange={this.handleChange}
                    searchList={this.state.list} />);
            }
            let internalWrapper;

            act(() => {
                internalWrapper = mount(<Test />);
            });

            act(() => {
                internalWrapper.find('.fd-input').simulate('click');
            });


            expect(document.querySelectorAll('li')).toHaveLength(searchData.length);

            act(() => {
                internalWrapper
                    .find(searchInput)
                    .simulate('change', { target: { value: 'pe' } });
            });

            expect(document.querySelectorAll('li')).toHaveLength(searchDataNew.length);

            act(() => {
                internalWrapper
                    .find(searchInput)
                    .simulate('change', { target: { value: searchDataNew[0].text } });
            });

            expect(document.querySelectorAll('li')).toHaveLength(1);
            internalWrapper.unmount();

        });
    });
});
