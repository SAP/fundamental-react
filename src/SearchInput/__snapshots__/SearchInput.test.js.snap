// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> create SearchInput 1`] = `
<div
  className="fd-search-input blue"
>
  <div
    className="fd-popover"
  >
    <div
      className="fd-popover__control"
    >
      <div
        aria-expanded={false}
        aria-haspopup="true"
        className="fd-combobox-control"
      >
        <div
          className="fd-input-group fd-input-group--after"
        >
          <input
            className="fd-input"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter a fruit"
            style={Object {}}
            type="text"
            value=""
          />
          <span
            className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
          >
            <button
              className=" fd-button--light sap-icon--search"
              onClick={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
    <div
      aria-hidden={true}
      className="fd-popover__body fd-popover__body--no-arrow"
    >
      <div
        className=""
      >
        <nav
          className="fd-menu"
        >
          <ul
            className="fd-menu__list"
          >
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              apple
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              apricot
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              banana
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              blueberry
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              blackberry
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              calabash
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              clementines
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              kiwi
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              orange
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SearchInput /> create SearchInput 2`] = `
<div
  className="fd-search-input fd-search-input--closed"
>
  <div
    className="fd-popover"
  >
    <div
      className="fd-popover__control fd-search-input__control"
    >
      <button
        aria-expanded={false}
        aria-haspopup="true"
        className="sap-icon--search fd-button--shell"
        onClick={[Function]}
      />
      <div
        aria-hidden={true}
        className="fd-search-input__closedcontrol"
      >
        <div
          aria-expanded={false}
          aria-haspopup="true"
          className="fd-search-input__controlinput"
        >
          <input
            className="fd-input"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter a fruit"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      aria-hidden={true}
      className="fd-popover__body fd-popover__body--no-arrow"
    >
      <div
        className="fd-search-input__body"
      >
        <nav
          className="fd-menu"
        >
          <ul
            className="fd-menu__list"
          >
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              apple
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              apricot
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              banana
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              blueberry
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              blackberry
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              calabash
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              clementines
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              kiwi
            </li>
            <li
              className="fd-menu__item"
              onClick={[Function]}
            >
              <strong>
                
              </strong>
              orange
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SearchInput /> create SearchInput 3`] = `
<div
  className="fd-search-input"
>
  <div
    className="fd-popover"
  >
    <div
      className="fd-popover__control"
    >
      <div
        aria-expanded={false}
        aria-haspopup="true"
        className="fd-combobox-control"
      >
        <div
          className="fd-input-group fd-input-group--after"
        >
          <input
            className="fd-input"
            onChange={[Function]}
            onClick={[Function]}
            onKeyPress={[Function]}
            placeholder="Enter a fruit"
            style={Object {}}
            type="text"
            value=""
          />
          <span
            className="fd-input-group__addon fd-input-group__addon--after fd-input-group__addon--button"
          >
            <button
              className=" fd-button--light sap-icon--search"
              onClick={[Function]}
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
